lib/literal/array.rb

```diff
@@ -64,6 +64,39 @@ def &(other)
 		end
 	end
 
+	def *(times)
+		case times
+		when String
+			@__value__ * times
+		else
+			__with__(@__value__ * times)
+		end
+	end
+
+	def +(other)
+		case other
+		when ::Array
+			other = Literal::Array(@__type__).new(*other)
+		when Literal::Array
+			# Do nothing
```

Comment:
Can we move line 86 in here?

Also, I think above this, we can handle the case that we’re adding an another Literal::Array of the same type. Should be able to do `when @__generic__` (see above).

Reply:
@joeldrapper Yes, it ends up looking either like: 
```ruby
def +(other)
  case other
  when ::Array
    other_literal = @__generic__.new(*other)
    Literal::Array.new(@__value__ + other_literal.__value__, type: @__type__)
  when Literal::Array
    Literal::Array.new(@__value__ + other.__value__, type: @__type__)
  else
    raise ArgumentError.new("Cannot perform + with #{other.class.name}.")
  end
end
```

or 

```ruby
def +(other)
  case other
  when ::Array
    values = @__value__ + @__generic__.new(*other).__value__
  when Literal::Array
    values = @__value__ + other.__value__
  else
    raise ArgumentError, "Cannot perform `+` with #{other.class.name}."
  end

  Literal::Array.new(values, type: @__type__)
end
```

I'm partial to the second, but let me know which style you prefer.